### PR TITLE
Join and Login form improvements

### LINKED ISSUE #99: Join and Login improvements
After you go through /schoolboard-conversation-candidate-recorder and get to the Thank You page, the ui needs some quick improvements:

The Name field needs to be required

The Login and Join buttons are tiny - and should be bigger and brigher
People are confused but having Login and Join combined in one thing - lets separate these in to what look like two tabs, where Join is the initial Tab, but you can hit the Login tab.  - but actually it's just this same component with two different 'shapes', and if you type your email and password on the join tab, and then hit the login tab, your email and password are still there, and you can switch back and forth.

Start by doing a rough sketch of the UI for the Join/Login component before implementing and lets get some feedback.   But we're trying to do quick and simple at this point.

### PR DESCRIPTION
**Pull request by luiscmartinez**
Completed Functionality. Working on ensure the form looks good on all screens. 
Closes #99

### CODE REVIEW

**Comment by luiscmartinez** [comment]
AuthForm is ready to be merged. 

### Join Form
image
* Form will display validation errors when email or password do not meet criteria.
*  User must check agreement terms before if they wish to join us
* First name and last name values aren't included in the signup request yet. 
### Validation messages
image


### Login Form
image
* When input fields are populated the button will become enabled and will change colors. 
### Active Btn
image


### Message for login errors
image


##### Any suggestions?
Should we keep the placeholder for email or remove it ?
I tried using the forgot password function but it didn't work right off the back. Should I look into it so I can include the forgot password button on login form. 
@ddfridley

**Comment by ddfridley** [comment]
on the join form if I enter everything but leave confirm password blank, when I hit the join button it blinks, but nothing happens and there's no error message.  I think previously the join button was not clickable until all fields were filled in.  In this case, I think there should be an error message saying confirm password.

**Comment by ddfridley** [comment]
on the forgot password question, it would be great to have it if you want to go down that path.  In development, it won't actually send out an email - because you haven't setup the env keys, but you should see a log message come out on the server saying what it would have sent and get the link from there.  If that doesn't work, lets look at it together.    But this is such an improvement I check this in and make another issue out of that.

File: app/components/auth-forms/index.jsx [hunk 1]
```diff
@@ -0,0 +1,262 @@
...
+      marginLeft: '0.5rem',
+      fontWeight: '800',
+    },
+  },
+  formValidationErrors: {
+    color: 'red',
+    width: '100%',
+  },
+}
+export const AuthForm = injectSheet(styles)(Temp)
```
Comment (ddfridley) [inline app/components/auth-forms/index.jsx:261]: this is causing injectSheet to be called twice, and creating 2 object, AuthForm and default.
I think you can do export default AuthForm and get the desired result.  You may also be able to say export default const AuthForm = ... but maybe that only works with classes.

File: app/components/web-components/undebate.jsx [hunk 2]
```diff
@@ -2456,12 +2456,8 @@ class RASPUndebate extends React.Component {
...
                             <span>Join and your recorded videos will be uploaded and shared</span>
                           </div>
-                          <div>
-                            <Join
-                              className={this.props.classes['join']}
-                              userInfo={{ name: this.state.name }}
-                              onChange={this.onUserLogin.bind(this)}
-                            ></Join>
+                          <div style={{ display: 'flex', justifyContent: 'center' }}>
+                            <AuthForm userInfo={{ name: this.state.name }} onChange={this.onUserLogin.bind(this)} />
```
Comment (ddfridley) [inline app/components/web-components/undebate.jsx:2460]: This is collecting firstName and lastName, if you add them to userInfo here they will get stored in the database.  You will also have to add firstName and lastName to app/models/user.js

I'm eager to see how you do it, I have not used react hooks but I want to do more.

**Comment by ddfridley** [comment]
sorry, one more comment.  the directory is called auth-forms but the component that's exported is AuthForm, and it's only returning one thing.  So I think the directory should be auth-form  what do you think, or were you thinking something else?

File: app/components/web-components/undebate.jsx (hunk 2 above)
Comment (ddfridley) [reply app/components/web-components/undebate.jsx:2460]: correction - it's in index.js handleLogin and handleSignUp that you need to add firstName and lastName to userInfo

Comment (luiscmartinez) [reply app/components/web-components/undebate.jsx:2460]: @ddfridley Will firstName and lastName be replacing the column "name" ?

**Comment by ddfridley** [comment]
We should add firstName and lastName and keep name.  -- name is what 
appears under the video.  Also Bria information has the combined field 
as well as the individual ones.

d

**Comment by epg323** [comment]
rate limiter times was set for a minute to make testing easier

**Comment by ddfridley** [comment]
I reviewed it and it's looking good - but since it now incorporates the

**Comment by ddfridley** [comment]
[I saw that firstName and lastName are going into the user record. thats great!]

In undebate.jsx, move the part that says "Name [Name ]This will be shown with your video"  to after the AuthForm, so that it does not appear until after one is logged in.  (It still shouldn't show if there is a bp_info.candidate_name).  Instead of saying Name, say "Name Shown with Video". 

If you try to join with an existing email, it says: "This email is already take" there needs to be an n at the end.

File: app/components/web-components/undebate.jsx [hunk 3]
```diff
@@ -2447,41 +2451,41 @@ class RASPUndebate extends React.Component {
...
-                              placeholder="Name"
-                              ref="name"
-                              name="name"
-                              onChange={e => this.setState({ name: e.value })}
-                            />
-                          </label>
-                          <span>This will be shown with your video</span>
-                        </div>
-                      ) : null}
+                      {!this.props.bp_info || !this.props.bp_info.candidate_name ? null : null}
```
Comment (luiscmartinez) [inline app/components/web-components/undebate.jsx:2454]: @ddfridley for now I have it render null no matter what. What should this be ?

**Comment by luiscmartinez** [comment]
image

After login or join form aka AuthForm !

**Comment by ddfridley** [comment]
closed by PR #195

**Pull request closed without merging**